### PR TITLE
[FL-3752] HID/BLE Keyboard UI refactoring

### PR DESCRIPTION
**Pull request by Astrrra**
# What's new

- See FL-3752

# Verification 

- Verify that the behavior and the visual changes match the ones described in the ticket

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: applications/system/hid_app/views/hid_keyboard.c [hunk 1]
```diff
@@ -210,17 +240,73 @@ static void hid_keyboard_draw_key(
...
+            MARGIN_LEFT + x * (KEY_WIDTH + KEY_PADDING) + keyWidth / 2 - key_icon->width / 2,
+            MARGIN_TOP + y * (KEY_HEIGHT + KEY_PADDING) + KEY_HEIGHT / 2 - key_icon->height / 2,
+            key_icon);
+
+        return;
+    }
+
+    if(key.key != 0) {
+        // Text with no shift
+        char key_str[2] = "\0\0";
```
Comment (hedger) [inline applications/system/hid_app/views/hid_keyboard.c]: = {key.key, '\0'}

File: applications/system/hid_app/views/hid_keyboard.c [hunk 2]
```diff
@@ -210,17 +240,73 @@ static void hid_keyboard_draw_key(
...
+        key_str[0] = key.key;
+
+        if(key.value >= HID_KEYBOARD_1 && key.value <= HID_KEYBOARD_0) {
+            // Special case for numbers, draw them one pixel lower
+            canvas_draw_str_aligned(
+                canvas,
+                MARGIN_LEFT + x * (KEY_WIDTH + KEY_PADDING) + keyWidth / 2 + 1,
+                MARGIN_TOP + y * (KEY_HEIGHT + KEY_PADDING) + KEY_HEIGHT / 2 + 1,
```
Comment (hedger) [inline applications/system/hid_app/views/hid_keyboard.c]: make a single call with optional value (+1/0) depending on if()

File: applications/system/hid_app/views/hid_keyboard.c [hunk 3]
```diff
@@ -210,17 +240,73 @@ static void hid_keyboard_draw_key(
...
+        return;
+    }
+    if(model->shift && key.key != 0) {
+        // Text and shift
         char key_str[2] = "\0\0";
+
         // If shift is toggled use the shift key when available
-        key_str[0] = (model->shift && key.shift_key != 0) ? key.shift_key : key.key;
+        key_str[0] = key.shift_key != 0 ? key.shift_key : key.key;
         canvas_draw_str_aligned(
```
Comment (hedger) [inline applications/system/hid_app/views/hid_keyboard.c:250]: wrap into a function taking (value, x, y, x_offs, y_offs)

**Review by hedger — DISMISSED** [review]

**Comment by hedger** [comment]
image

image


1. Lower modifier keys row text is not vertically aligned (1 px) when not pressed
2. Pressed shift key is not aligned
3. `/ \ '` keys not affected by shift

**Comment by Astrrra** [comment]
@hedger 1 and 2 are intentional (to make buttons look like they're actually pressed down), 3 has been fixed

**Review by skotopes — APPROVED** [review]

**Pull request merged**